### ISSUE TITLE
Theano Error with Ubuntu 14.04: Assertion `PyArray_NDIM(V5) == 1' failed.

### ISSUE THREAD

**Issue by tywp1115** [Theano/Theano#3735]
I am running off of Ubuntu 14.04

I am using Keras to create a neural network. My code has a threading.Timer running and every 10 seconds, it uses the Keras model on some given data. (Keras uses Theano).

It will run for several hours before running into this issue. (There are no memory leaks).

Why is this happening? I am trying to run off a CPU (but I could run off a GPU). Am I accidently feeding in NaN values to the Keras model? I cant find any other similar question online.    

I manually set these because it gave me a warning that I should
    theano.config.optimizer = "fast_compile"
    theano.config.exception_verbosity = "high"

This is the error
    python: /home/ty/.theano/compiledir_Linux-3.19--generic-x86_64-with-Ubuntu-14.04-trusty-x86_64-2.7.6-64/tmpqvZZMx/mod.cpp:111: int {anonymous}::__struct_compiled_op_943369c7c09a75f89b7161200d6ceb5a::run(): Assertion `PyArray_NDIM(V5) == 1' failed.
Aborted (core dumped)

**Comment by nouiz** [comment]
Can you provide the full error message? It contain extra information.

Do you use multiple threads? Do they share the same Theano function? This
isn't will supported. You can't safely have 2 threads u use the same Theano
function at the same time.

Fred

**Comment by tywp1115** [comment]
That is the entire message.

I am using a threading.Timer(...) to run a theano function every 5s. It's possible that the function wont finish in that 5s but highly unlikely. (Im using theano as a backend for Keras - for Neural Networks).

If Keras (or Theano) doesnt finish within that 5s, I want to stop the theano function (and restart it with the new keras.model.predict(...)). 

TLDR: How can I stop theano to prevent a threading error? Is this issue with Keras or with Theano?

**Comment by daqulazhang** [comment]
any progress on this? We met this issue too.. and we are sure only one thread called theano function in one time/

**Comment by tywp1115** [comment]
No luck.

**Comment by lamblin** [comment]
Theano is not thread-safe, and we have no plans to make it thread-safe at the moment.
If you are executing on GPU, then the asynchronous execution of GPU kernels means that even when the call to the Theano function returns, there may still be computation and memory access going on on the GPU.
Please consider using processes rather than threads.
